modules/aruco/include/opencv2/aruco_detector.hpp

```diff
@@ -0,0 +1,441 @@
+      * thresholding (otherwise, all the bits are set to 0 or 1 depending on mean higher than 128 or not) (default 5.0)
+      */
+    CV_PROP_RW double minOtsuStdDev;
+
+    /// error correction rate respect to the maximun error correction capability for each dictionary (default 0.6).
+    CV_PROP_RW double errorCorrectionRate;
+
+    /** @brief April :: User-configurable parameters.
+      * detection of quads can be done on a lower-resolution image, improving speed at a cost of
+      * pose accuracy and a slight decrease in detection rate. Decoding the binary payload is still
+      */
+    CV_PROP_RW float aprilTagQuadDecimate;
+
+    /// what Gaussian blur should be applied to the segmented image (used for quad detection?)
+    CV_PROP_RW float aprilTagQuadSigma;
+
+    // April :: Internal variables
+    /// reject quads containing too few pixels (default 5).
+    CV_PROP_RW int aprilTagMinClusterPixels;
+
+    /// how many corner candidates to consider when segmenting a group of pixels into a quad (default 10).
+    CV_PROP_RW int aprilTagMaxNmaxima;
+
+    /** @brief reject quads where pairs of edges have angles that are close to straight or close to 180 degrees.
+      * Zero means that no quads are rejected. (In radians) (default 10*PI/180)
+      */
+    CV_PROP_RW float aprilTagCriticalRad;
+
+    /// when fitting lines to the contours, what is the maximum mean squared error
+    CV_PROP_RW float aprilTagMaxLineFitMse;
+
+    /** @brief when we build our model of black & white pixels, we add an extra check that the
+      * white model must be (overall) brighter than the black model.  How much brighter? (in pixel values, [0,255]).
+      * (default 5)
+      */
+    CV_PROP_RW int aprilTagMinWhiteBlackDiff;
+
+    /// should the thresholded image be deglitched? Only useful for very noisy images (default 0).
+    CV_PROP_RW int aprilTagDeglitch;
+
+    /** @brief to check if there is a white marker. In order to generate a "white" marker just invert a
+      * normal marker by using a tilde, ~markerImage. (default false)
+      */
+    CV_PROP_RW bool detectInvertedMarker;
+
+    /** @brief new Aruco functionality proposed in the paper:
+      * Romero-Ramirez et al: Speeded up detection of squared fiducial markers (2018)
+      * https://www.researchgate.net/publication/325787310_Speeded_Up_Detection_of_Squared_Fiducial_Markers
+      */
+
+    /// to enable the new and faster Aruco detection strategy.
+    CV_PROP_RW bool useAruco3Detection;
+
+    /// minimum side length of a marker in the canonical image. Latter is the binarized image in which contours are searched.
+    CV_PROP_RW int minSideLengthCanonicalImg;
+
+    /// range [0,1], eq (2) from paper. The parameter tau_i has a direct influence on the processing speed.
+    CV_PROP_RW float minMarkerLengthRatioOriginalImg;
+
+private:
+    bool readWrite(const Ptr<FileNode>& readNode = nullptr, const Ptr<FileStorage>& writeStorage = nullptr);
```

Comment:
Why do we need this in public OpenCV API?

Reply:
removed

Reply:
@AleksandrPanov It's still there.

Reply:
removed `readWrite()` from public API